src/wazuh_modules/syscollector/src/syscollectorImp.cpp

```diff
@@ -1476,11 +1476,11 @@ nlohmann::json Syscollector::getPortsData()
     constexpr auto PORT_LISTENING_STATE { "listening" };
     constexpr auto TCP_PROTOCOL { "tcp" };
     constexpr auto UDP_PROTOCOL { "udp" };
-    const auto& data { m_spInfo->ports() };
+    auto data { m_spInfo->ports() };
```

Comment:
This change has generated the errors in the Checks, you can see it as `[json.exception.type_error.304] cannot use at() with array`.

The thing is that the expression `auto data { m_spInfo->ports() };` will generate a new array and inside it will store the result from the function that already is an array.

Previously, the expression `const auto& data { m_spInfo->ports() };`  was a constant reference to the array returned by the method and no exception was generated.

See the following examples.

### Code 1

```c++
nlohmann::json function() {
  auto array = nlohmann::json::array();
  array.push_back({{"name", "test"}});
  return array;
}

int main() {
  const auto &data{function()};

  if (data.is_array()) {
    std::cout << "Data is an array" << std::endl;
  } else {
    std::cout << "Data is not an array" << std::endl;
  }
  std::cout << "Data: " << data << std::endl;

  for (auto &item : data) {
    if (item.is_array()) {
      std::cout << "Item is an array" << std::endl;
    } else {
      std::cout << "Item is not an array" << std::endl;
    }
    std::cout << "Item: " << item << std::endl;

    item.at("name");
  }

  return 0;
}
```

**Output**

```
Data is an array
Data: [{"name":"test"}]
Item is not an array
Item: {"name":"test"}
``` 

### Code 2

```c++
nlohmann::json function() {
  auto array = nlohmann::json::array();
  array.push_back({{"name", "test"}});
  return array;
}

int main() {
  auto data{function()}; // Only changed this expression 

  if (data.is_array()) {
    std::cout << "Data is an array" << std::endl;
  } else {
    std::cout << "Data is not an array" << std::endl;
  }
  std::cout << "Data: " << data << std::endl;

  for (auto &item : data) {
    if (item.is_array()) {
      std::cout << "Item is an array" << std::endl;
    } else {
      std::cout << "Item is not an array" << std::endl;
    }
    std::cout << "Item: " << item << std::endl;

    item.at("name");
  }

  return 0;
}
```

**Output**

```
Data is an array
Data: [[{"name":"test"}]]
Item is an array
Item: [{"name":"test"}]
terminate called after throwing an instance of 'nlohmann::detail::type_error'
  what():  [json.exception.type_error.304] cannot use at() with array
[1]    201573 IOT instruction (core dumped)  ./test
```

Reply:
But another question: why did you change the reference for a local variable?

Reply:
The report from Coverity only makes a reference to the **auto** in the loop as far as I can see.
So the change in the line 1479 might not be required (although it's suggested in the issue)

![coverity](https://user-images.githubusercontent.com/65046601/206002060-7f7691ac-c1cd-4476-82a1-ac9371ca57a5.png)

Reply:
yes sure, but `item` is related to `data`, internally in the loop `item` is being modified so the original object would be modified